### PR TITLE
NP-766 Version bump datamodel to v0.5.1

### PR DESCRIPTION
**Pull request by nraanes**

### CODE REVIEW

File: src/main/java/no/unit/nva/doi/transformer/DataciteResponseConverter.java [hunk 1]
```diff
@@ -115,7 +116,10 @@ private PublicationSubtype createPublicationSubType() {
     }
 
     private PublicationType extractPublicationType(DataciteResponse dataciteResponse) {
-        return PublicationType.lookup(dataciteResponse.getTypes().getResourceType());
+        String dataciteResourceType =
+                dataciteResponse.getTypes().getResourceType().equalsIgnoreCase(JOURNAL_ARTICLE.getValue())
```
Comment (axthosarouris) [inline src/main/java/no/unit/nva/doi/transformer/DataciteResponseConverter.java]: This is too long. break it down

File: src/test/java/no/unit/nva/doi/transformer/CrossRefConverterTest.java [hunk 2]
```diff
@@ -227,7 +230,7 @@ public void toPublicationSetsThePagesOfTheReferenceWhentheCrossrefDocHasPages()
         Pages actualPages = toPublication(sampleInputDocument).getEntityDescription().getReference()
                                                               .getPublicationInstance()
                                                               .getPages();
-        Pages expectedPages = new Pages.Builder().withBegins("45").withEnds("89").build();
+        Pages expectedPages = new Range.Builder().withBegin("45").withEnd("89").build();;
```
Comment (axthosarouris) [inline src/test/java/no/unit/nva/doi/transformer/CrossRefConverterTest.java]: double semicolon

**Review by brinxmat — DISMISSED** [review]

File: build.gradle [hunk 3]
```diff
@@ -19,7 +19,7 @@ targetCompatibility = JavaVersion.VERSION_11 // bytecode target version
 def junit5Version = '5.6.0'
 
 dependencies {
-    implementation group: 'com.github.BIBSYSDEV', name: 'nva-datamodel-java', version: 'v0.4.0'
+    implementation group: 'com.github.BIBSYSDEV', name: 'nva-datamodel-java', version: 'v0.5.1'
     implementation group: 'com.amazonaws', name: 'aws-lambda-java-core', version: '1.2.0'
     implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.10.2'
```
Comment (janbj) [inline build.gradle:24]: You can use a variable for versions that are the same

File: src/main/java/no/unit/nva/doi/transformer/CrossRefConverter.java [hunk 4]
```diff
@@ -124,33 +127,45 @@ private boolean containsCrossrefAsSource(CrossRefDocument document) {
...
     }
 
+    private PublicationContext extractPublicationContext(CrossRefDocument document) {
+        try {
+            return new Journal.Builder()
+                .withLevel(null)
+                .withTitle(extractJournalTitle(document))
+                .build();
+        } catch (InvalidIssnException e) {
+            e.printStackTrace();
```
Comment (janbj) [inline src/main/java/no/unit/nva/doi/transformer/CrossRefConverter.java:150]: You should use the LambdaLogger and provide some context for what's failing.

File: src/main/java/no/unit/nva/doi/transformer/CrossRefConverter.java [hunk 5]
```diff
@@ -210,14 +225,38 @@ private String extractTitle(CrossRefDocument document) {
...
-    private Contributor toContributor(Author author, int alternativeSequence) {
+    private List<Contributor> successfulMappings(List<Try<Contributor>> contributorMappings) {
+        return contributorMappings.stream()
+                                  .filter(Try::isSuccess)
+                                  .map(Try::get)
+                                  .collect(Collectors.toList());
+    }
+
+    private void reportFailures(List<Try<Contributor>> contributors) {
+        contributors.stream().filter(Try::isFailure).forEach(failure -> failure.getException().printStackTrace());
```
Comment (janbj) [inline src/main/java/no/unit/nva/doi/transformer/CrossRefConverter.java:249]: Should probably use the Logger and not print stack traces directly.

**Review by janbj — DISMISSED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**